includes/REST_API/Embed_Controller.php

```diff
@@ -168,15 +168,47 @@ public function get_proxy_item( $request ) {
 	/**
 	 * Retrieves the story metadata for a given URL on the current site.
 	 *
-	 * @SuppressWarnings(PHPMD.NPathComplexity)
-	 *
 	 * @since 1.0.0
 	 *
 	 * @param string $url  The URL that should be inspected for metadata.
 	 *
 	 * @return array|false Story metadata if the URL does belong to the current site. False otherwise.
 	 */
 	private function get_data_from_post( $url ) {
+		$post_id = $this->url_to_post_id( $url );
+
+		if ( ! $post_id ) {
+			return false;
+		}
+
+		$post = get_post( $post_id );
```

Comment:
How does this work it the post comes from another site on a multisite?

Think of this 
Main site: www.example.com and second site www.example.com/test. 
1. On www.example.com you embed, www.example.com/test/web-stories/test. 
2. Logic gets post id 123 on site 2. 
3. Get to this line, it would get post id from 123 from site 1. 

Am I missing something here?

Reply:
There‘s some multisite switching already in place, I am not changing that. The only new part is the regex.

Reply:
Again, the whole logic here for multisite already existed, and is mostly a copy of `get_oembed_response_data_for_url()`. I didn't change it here. It tries switching to the sub-site based on the URL, and get the post ID afterwards.